include/mls/credential.h

```diff
@@ -2,6 +2,9 @@
 
 #include "mls/common.h"
 #include "mls/crypto.h"
+#include <memory>
+#include <openssl/x509.h>
+#include <openssl/x509v3.h>
```

Comment:
Let's not use OpenSSL constructs in the public API for this.  I would like to at least maintain the pretense that we are independent of the underlying crypto library, even if we use OpenSSL internally, and callers are likely to use it to process what we provide.